crates/c-api/include/wasmtime/store.h

```diff
@@ -79,6 +79,42 @@ WASM_API_EXTERN wasmtime_store_t *wasmtime_store_new(
  */
 WASM_API_EXTERN wasmtime_context_t *wasmtime_store_context(wasmtime_store_t *store);
 
+/**
+ * \brief Provides limits for a store. Used by hosts to limit resource
+ * consumption of instances. Use `0` to keep the default value for the limit.
+ *
+ * \param store store where the limits should be set.
+ * \param memory_size the maximum number of bytes a linear memory can grow to.
+ * Growing a linear memory beyond this limit will fail. By default,
+ * linear memory will not be limited.
+ * \param table_elements the maximum number of elements in a table.
+ * Growing a table beyond this limit will fail. By default, table elements
+ * will not be limited.
+ * \param instances the maximum number of instances that can be created
+ * for a Store. Module instantiation will fail if this limit is exceeded.
+ * This value defaults to 10,000.
+ * \param tables the maximum number of tables that can be created for a Store.
+ * Module instantiation will fail if this limit is exceeded. This value
+ * defaults to 10,000.
+ * \param memories the maximum number of linear memories that can be created
+ * for a Store. Instantiation will fail with an error if this limit is exceeded.
+ * This value defaults to 10,000.
+ *
+ * Use `0` for the parameters that should be kept on default values.
```

Comment:
I think there's a case to be made for allowing precisely 0 of the above items, so could this instead perhaps use a negative value to indicate "use the default"?

Reply:
I was thinking about this as well, but in that case we would need to use different types then `usize` and `u32`. What do you suggest using instead of these?

Reply:
For some of these values the range of `0..i32::MAX` is sufficient (or even `isize::MAX`). Otherwise though you could also add a dedicated `struct` with getters/setters where not calling a setter is equivalent to "none" or something like that. I don't know the best way to handle optional parameters in C myself.

Reply:
`struct` is not perfect, you'd end up with same problems, even if you use pointers on integers.. because then (afaik) you wouldn't be able to distinguish `0` and `NULL`.

Let's go with `-1` as indication "to leave default" and `i64` for all types. That way we keep the ranges "sort of same" and we will have a way to use `0` as a correct value.